core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java

```diff
@@ -806,11 +806,13 @@ private StatedImage getCustomStatedImage(ImageCatalog imageCatalog, String image
             StatedImage sourceImage = getSourceImageByImageType(customImage);
             LOGGER.info("Custom image '{}' is a {} image '{}' customization.", imageId, customImage.getImageType(), customImage.getCustomizedImageId());
             return customImageProvider.mergeSourceImageAndCustomImageProperties(
-                        sourceImage, customImage, imageCatalog.getImageCatalogUrl(), imageCatalog.getName());
+                    sourceImage, customImage, imageCatalog.getImageCatalogUrl(), imageCatalog.getName());
+        } else if (Strings.isNullOrEmpty(imageCatalog.getImageCatalogUrl())) {
+            throw new CloudbreakImageNotFoundException(
+                    String.format("Could not find any custom image with id: '%s' in catalog: '%s'", imageId, imageCatalog.getName()));
```

Comment:
DB based custom image catalogs don't have a URL by nature. If the image is within such catalog, the error message is problematic - imageID will match, catalog name will match, yet the response will still say "could not find any image", which is confusing.

Reply:
Already discussed on slack:
It is not related to the bug but unit tests highlighted that null imageCatalogUrl will cause an NPE. This is a generic public method to lookup an image so having this error message makes sense.

Reply:
In terms of changing the image catalog, this method is used to validate that the target image catalog includes the image (by uuid) what we already use from the source catalog.